Class-19-(Authentication).md

```diff
@@ -0,0 +1,25 @@
+# Class 19: Authentication
+
+## In the prep work this week, the students learned
+
+1. What authentication is.
+1. What sessions and cookies are and how sessions and cookies are used to verify a user's identity.
+1. The importance of encryption.
+1. Why developers use one-way encryption to protect users' passwords.
+1. How to use Spring Security in an application to enable authentication.
+
+## Announcements
+
+1. Students are almost done with the class! One class left!
+1. Assignment #4 is due next class! 
+1. Check with your course manager for any additional announcements.
+
+## Large Group Time (Instructor)
+
+### Class 19 Topics That Require Careful Attention
+1. Review hashing and one-way encryption. Remind students that passwords are not stored, but the encrypted password is stored.
+1. 
+
+## Small Group Time: Class 19 Studio (TA Notes)
+
+1. The studio is focused on adding authentication to ``coding-events``.
```

Comment:
We don't have a studio at this point. We talked about potentially having them add authentication to TechJobs, but unless they have finished the assignment that won't be a great idea. We should revisit this as a group and decide what to do.

Reply:
Added some tips and tricks!